container/docker/handler.go

```diff
@@ -242,17 +273,22 @@ func (self *dockerContainerHandler) getFsStats(stats *info.ContainerStats) error
 	if self.ignoreMetrics.Has(container.DiskUsageMetrics) {
 		return nil
 	}
+	var device string
 	switch self.storageDriver {
+	case devicemapperStorageDriver:
```

Comment:
@vishh 

> Why are you adding the fs logic here, instead of abstracting all of it in FsHandler?

I added it here because I had thought the substance of our agreement was to isolate knowledge of device mapper to the docker driver.  The Fs handler we use is common code, and I thought we discussed 1) keeping device mapper out of the common fs handler 2) avoiding creating a new fs handler that was similar to the common one, but also knew about device mapper.

If you would prefer, I can do one of the following:

1.  Create a distinct docker fs handler that took a thin pool watcher and a common fs handler and composed together the usage
2.  Add a thin pool watcher to the common fs handler and just handle all of it in the common fs handler
3.  Something else